pandas/io/pytables.py

```diff
@@ -1133,6 +1133,10 @@ def _create_storer(self, group, format=None, value=None, append=False,
                        **kwargs):
         """ return a suitable class to operate """
 
+        if isinstance(value, Series) and any(np.iscomplex(value)) and format=='table':
```

Comment:
I don' t think `set_atom` can tell if it is a `Series` since the data is not a `block`.